site-user-management/layouts/api_user_deletion.tpl

```diff
@@ -176,38 +172,38 @@
           }
         }
 
-        const getUsers = (pageNumber) => new Promise((resolve, reject) => {
+        const getUsers = (pageNumber = 1) => new Promise((resolve, reject) => {
           $.ajax({
               type: 'GET',
               contentType: 'application/json',
               url: `/admin/api/site_users?s=site_user.email&per_page=250&page=${pageNumber}`,
               dataType: 'json',
             })
             .then((users, status, xhr) => {
-              const nPages = parseInt(xhr.getResponseHeader('X-Total-Pages'));
-              const obj = {users: users, pages: nPages}
-              buildTable(users);
-              resolve(obj);
+              const pages = parseInt(xhr.getResponseHeader('X-Total-Pages'));
+
+              if (pageNumber === 1 && pageNumber != pages) {
+                $('#user-table').empty();
+                addToTable(users);
+                pageNumber += 1;
+                getUsers(pageNumber);
+              } else if (pageNumber > pages) {
+                $('#user-table-table').append('</tbody></table>');
+              } else {
+                pageNumber += 1;
+                addToTable(users);
+                getUsers(pageNumber);
+              }
```

Comment:
Siin on kaks teemat segamini läinud. Funktsioon peaks tegema ühte asja, tegema seda hästi ja vastama oma nimele. Ehk siin ei peaks üldse DOMi puutuma ja kogu tabeli konstrueerimise `addToTable` hooleks jätma. Viimasele võid ette anda parameetri a la `init` või `empty` et see tühjendada. ` $('#user-table-table').append('</tbody></table>')` rida ja vastav `if` haru on üldse ebavajalik, sest see peaks alati DOM-is olemas olema ning muudad sa ainult `tbody` sisu.

Teiseks teed siin üleliigseid päringuid. Näiteks kui lk-de arv on 1, siis tehakse algne päring, kutsutakse välja `addToTable` ja tehakse järgmine (üleliigne) päring, mis saab tühja vastuse. Lisaks on siin probleem, et kui lk-de arv on 1, ei tühjendata algset tabelit. Kuid 1 üleliigse päringu probleem on suvalise lk-de arvu puhul.

Samuti ärme kasuta `!=`, v.a. erijuhul kui tahame mingit väärtust `null` ja `undefined` vastu [korraga kontrollida](https://stackoverflow.com/a/15992131).

`pageNumber += 1` on üleliigne, saad otse ette anda `getUsers(pageNumber + 1)`.

`pages` võiks selguse mõttes nimetada `nPages`. `pages` viitab, et muutujas hoitakse mingit lehekülgi esitavat struktuuri.

Kui tahad API päringud jadamisi toimivaks jätta, siis loogika on lihtsalt (`then` callbackis):

```
if (page === 1) {
  addToTable(users, {init: true});
}
else {
  addToTable(users);
}

if (nPages > pageNumber) {
  getUsers(pageNumber + 1);
}
```

või veel kompaktsemalt: 

```
addToTable(users, {init: page === 1});

if (nPages > pageNumber) {
  getUsers(pageNumber + 1);
}
```

Aga proovi ka paralleelseks teha, nagu Slackis arutatud.